gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaStreamingExtractor.java

```diff
@@ -214,6 +220,19 @@ public LongWatermark getLwm() {
 
   public KafkaStreamingExtractor(WorkUnitState state) {
     super(state);
+    this.topicPartitions = getTopicPartitionsFromWorkUnit(state);
+    Map<KafkaPartition, LongWatermark> topicPartitionWatermarks = getTopicPartitionWatermarks(this.topicPartitions);
+    if (this.maxAvgRecordSize > 0 ) {
+      long maxPollRecords =
+          state.getPropAsLong(MAX_KAFKA_BUFFER_SIZE_IN_BYTES, DEFAULT_MAX_KAFKA_BUFFER_SIZE_IN_BYTES) / maxAvgRecordSize;
```

Comment:
What's the intuition behind a 50 MB default max buffer size?

Reply:
Does it have to do with
```
source.kafka.consumerConfig.receive.buffer.bytes=655360
```

in gobblin-kafka-jobs?

Reply:
there is no relationship between those two config.  The value should not be super small as we need to guarantee the performance, and not super large as we want to avoid OOM issue. So I gave 50MB here as default. And this should only take effect when record size are large, for normal topic, it should fall back to use KAFKA_MAX_POLL_RECORDS_KEY

Reply:
Makes sense. I think 50MB seems like a good default. 

Here's an example of how the throughput would look for some higher volume topics. All topics under 5kb would be able to do 1000 records per second (pageviewevent is only ~800bytes and URE is ~3MB).